standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStore.java

```diff
@@ -573,7 +573,7 @@ public void init() throws MetaException {
       listeners.add(new SessionPropertiesListener(conf));
       transactionalListeners = MetaStoreServerUtils.getMetaStoreListeners(TransactionalMetaStoreEventListener.class,
           conf, MetastoreConf.getVar(conf, ConfVars.TRANSACTIONAL_EVENT_LISTENERS));
-      transactionalListeners.add(new AcidEventListener(conf));
+      transactionalListeners.add(0, new AcidEventListener(conf));
```

Comment:
Since you already know this has to be at the 0th index, you can add this add the beginning instead of shifting all other listeners and adding this to 0th index later.

Reply:
Also how do you make sure that notification event listener is added to the end?

Reply:
Good to add a test for the same

Reply:
It is small list and the moving will not matter. The current code MetaStoreServerUtils.getMetaStoreListeners() returns empty or a list of listener. The code looks clean this way.

Reply:
Can you suggest the test what you are expecting to cover?

Reply:
Test to make sure notification listener is the last listener. How do you make sure that notification event listener is added to the end in current code? 
The listener list maybe small now but can grow in the future. Wouldn't recommend adding a known overhead.